examples/s2_field_combination_extern_rf_train/nrw_crop_extern_s2_workflow_to_datasets_with_timeshift.ipynb

```diff
@@ -0,0 +1,2364 @@
+    "    print(sentinel2_band_workflow_dataset_task.get_status())\n",
+    "    status_download[b] = sentinel2_band_workflow_dataset_task.get_status()\n",
+    "\n",
+    "status_download"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Check the quota used for downloading the bands:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "0"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "used_quota_download = ge.get_quota(user_id)['used'] - used_quota_start\n",
+    "used_quota_download"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Monthly, cloud-free aggregations of the Sentinel-2 bands & NDVI\n",
+    "\n",
+    "For the training, we use the Sentinel-2 data of the bands 02, 03, 04, and 08. The scene classification layer (SCL) is used to filter out cloudy pixels for each band using an expression. The NDVI is calculated using an expression on band 4 and 8. \n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the bands dataset_ids, we create a map from band name to Geo Engine \"InternalDataId\". The \"InternalDataId\" is used to reference the datasets in the workflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'B02': {'type': 'internal',\n",
+       "  'datasetId': 'a0630d89-fe7d-4087-98e4-aa496e681d50'},\n",
+       " 'B03': {'type': 'internal',\n",
+       "  'datasetId': '1cfd9d51-6cbf-4490-9ce6-da224a6f5a83'},\n",
+       " 'B04': {'type': 'internal',\n",
+       "  'datasetId': '6441f1f3-a9fb-4298-9161-eab91f0825e2'},\n",
+       " 'B08': {'type': 'internal',\n",
+       "  'datasetId': '66479fef-4295-48f1-99bf-670f6b7ea540'},\n",
+       " 'SCL': {'type': 'internal',\n",
+       "  'datasetId': '4d243f03-f59b-4298-80e9-0fb5aa94926e'}}"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## IF you ran the workflow, you can get the ids from the status\n",
+    "band_data_map = { name: task_status.info['dataset'] for name, task_status in status_download.items() }\n",
+    "## IF you have the data already added, youn simply define the ids.\n",
+    "#band_data_map = {\n",
+    "#    'B02': 'a0630d89-fe7d-4087-98e4-aa496e681d50',\n",
+    "#    'B03': '1cfd9d51-6cbf-4490-9ce6-da224a6f5a83',\n",
+    "#    'B04': '6441f1f3-a9fb-4298-9161-eab91f0825e2',\n",
+    "#    'B08': '66479fef-4295-48f1-99bf-670f6b7ea540',\n",
+    "#    'SCL': '4d243f03-f59b-4298-80e9-0fb5aa94926e'\n",
```

Comment:
How do you know the ids, they weren't defined anywhere?

Reply:
i just put them there from the prev download task.